src/audio/audio_manager.cpp

```diff
@@ -16,17 +21,74 @@ AudioManager::AudioManager() : muted(true) {
         if (!music.openFromFile(filename)) {
             std::cerr << "AudioManager: Cannot open " << filename << std::endl;;
         }
-        music.setLoop(true);
-        if (!muted)
+
+        music.setVolume(d.volume);
+        music.setLoop(d.loop);
+
+        if (!muted) {
```

Comment:
Okay, makes sense. However, I still prefer `if (muted) return; music.play();` over `if (!muted) music.play();` since it's less nesting and makes the intention clear to not do anything after this point if muted.